bedboss/bedstat/bedstat.py

```diff
@@ -0,0 +1,229 @@
+import pypiper
+import bbconf
+
+
+SCHEMA_PATH_BEDSTAT = os.path.join(
+    os.path.dirname(os.path.realpath(__file__)), "pep_schema.yaml"
+)
+
+
+def hash_bedfile(filepath: str) -> str:
+    """
+    Generate digest for bedfile
+    :param str filepath: path to the bed file
+    :return str: digest of the files
+    """
+    with gzip.open(filepath, "rb") as f:
+        # concate column values
+        chrs = ",".join([row.split()[0].decode("utf-8") for row in f])
+        starts = ",".join([row.split()[1].decode("utf-8") for row in f])
+        ends = ",".join([row.split()[2].decode("utf-8") for row in f])
+        # hash column values
+        chr_digest = md5(chrs.encode("utf-8")).hexdigest()
+        start_digest = md5(starts.encode("utf-8")).hexdigest()
+        end_digest = md5(ends.encode("utf-8")).hexdigest()
+        # hash column digests
+        bed_digest = md5(
+            ",".join([chr_digest, start_digest, end_digest]).encode("utf-8")
+        ).hexdigest()
+
+        return bed_digest
+
+
+def convert_unit(size_in_bytes: int) -> str:
+    """
+    Convert the size from bytes to other units like KB, MB or GB
+    :param int size_in_bytes: size in bytes
+    :return str: File size as string in different units
+    """
+    if size_in_bytes < 1024:
+        return str(size_in_bytes) + "bytes"
+    elif size_in_bytes in range(1024, 1024 * 1024):
+        return str(round(size_in_bytes / 1024, 2)) + "KB"
+    elif size_in_bytes in range(1024 * 1024, 1024 * 1024 * 1024):
+        return str(round(size_in_bytes / (1024 * 1024))) + "MB"
+    elif size_in_bytes >= 1024 * 1024 * 1024:
+        return str(round(size_in_bytes / (1024 * 1024 * 1024))) + "GB"
+
+
+def bedstat(
+    bedfile: str,
+    bedbase_config: str,
+    genome: str,
+    outfolder: str,
+    ensdb: str = None,
+    open_signal_matrix: str = None,
+    bigbed: str = None,
+    sample_yaml: str = None,
+    just_db_commit: bool = False,
+    no_db_commit: bool = False,
+    force_overwrite: bool = False,
+    pm: pypiper.PipelineManager = None,
+    **kwargs,
+) -> NoReturn:
+    """
+    Run bedstat pipeline. Can be used without running from command line
+    :param str bedfile: a full path to bed file to process
+    :param str bigbed: a full path to bigbed
+    :param str bedbase_config: a path to the bedbase configuration file
+    :param str open_signal_matrix: a full path to the openSignalMatrix
+        required for the tissue specificity plots
+    :param str outfolder: folder for storing results of the pipeline
+    :param str genome: genome assembly of the sample
+    :param str ensdb: a full path to the ensdb gtf file required for genomes
+        not in GDdata
+    :param str sample_yaml: a yaml config file with sample attributes to pass
+        on more metadata
+        into the database
+    :param bool just_db_commit: whether just to commit the JSON to the database
+    :param bool no_db_commit: whether the JSON commit to the database should be
```

Comment:
how these work is confusing to me. do we need 2 options here, or can it be a true/false? 

and what is "the JSON" ? add a bit more detail

Reply:
IDK, It's also a bit confusing for me. I think Michał implemented this way... So I thought it's right way to do